core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -865,6 +865,15 @@ protected void convertOrder(
     }
   }
 
+  /**
+   * Returns whether we should keep the sort for the subsequent query conversion.
+   *
+   * @param topQuery Whether the query is in the top level.
+   */
+  boolean keepSortInSubQuery(boolean topQuery) {
+    return config.isKeepSortInSubQuery() || topQuery;
```

Comment:
Is there any special reason we made this function `protected` scope ?

Reply:
No, I'll make it private.